jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/MultiPart.java

```diff
@@ -407,7 +407,13 @@ public Content.Source newContentSource()
                 // because the content sources may not be read, or their chunks could be
                 // further retained, so those chunks must not be linked to the original ones.
                 List<Content.Chunk> chunks = content.stream()
-                    .map(chunk -> Content.Chunk.from(chunk.getByteBuffer().slice(), chunk.isLast()))
+                    .map(chunk ->
+                    {
+                        // TODO is this a reliable way to duplicate a chunk while eventually severing it from its retainable?
```

Comment:
Did we give up on the idea of trailer chunks, and other kind of non-content, non-failure, non-EOF, non-EMPTY buffers?

If we did, this comment can go away, otherwise this code won't correctly duplicate the chunk.